nav2_behavior_tree/include/nav2_behavior_tree/bt_cancel_action_node.hpp

```diff
@@ -0,0 +1,183 @@
+    const BT::NodeConfiguration & conf)
+  : BT::ActionNodeBase(xml_tag_name, conf), action_name_(action_name)
+  {
+    node_ = config().blackboard->template get<rclcpp::Node::SharedPtr>("node");
+    callback_group_ = node_->create_callback_group(
+      rclcpp::CallbackGroupType::MutuallyExclusive,
+      false);
+    callback_group_executor_.add_callback_group(callback_group_, node_->get_node_base_interface());
+
+    // Get the required items from the blackboard
+    bt_loop_duration_ =
+      config().blackboard->template get<std::chrono::milliseconds>("bt_loop_duration");
+    server_timeout_ =
+      config().blackboard->template get<std::chrono::milliseconds>("server_timeout");
+    getInput<std::chrono::milliseconds>("server_timeout", server_timeout_);
+
+    std::string remapped_action_name;
+    if (getInput("server_name", remapped_action_name)) {
+      action_name_ = remapped_action_name;
+    }
+    createActionClient(action_name_);
+
+    // Give the derive class a chance to do any initialization
+    RCLCPP_DEBUG(
+      node_->get_logger(), "\"%s\" BtCancelActionNode initialized",
+      xml_tag_name.c_str());
+  }
+
+  BtCancelActionNode() = delete;
+
+  virtual ~BtCancelActionNode()
+  {
+  }
+
+  /**
+   * @brief Create instance of an action client
+   * @param action_name Action name to create client for
+   */
+  void createActionClient(const std::string & action_name)
+  {
+    // Now that we have the ROS node to use, create the action client for this BT action
+    action_client_ = rclcpp_action::create_client<ActionT>(node_, action_name, callback_group_);
+
+    // Make sure the server is actually there before continuing
+    RCLCPP_DEBUG(node_->get_logger(), "Waiting for \"%s\" action server", action_name.c_str());
+    action_client_->wait_for_action_server();
+  }
+
+  /**
+   * @brief Any subclass of BtCancelActionNode that accepts parameters must provide a
+   * providedPorts method and call providedBasicPorts in it.
+   * @param addition Additional ports to add to BT port list
+   * @return BT::PortsList Containing basic ports along with node-specific ports
+   */
+  static BT::PortsList providedBasicPorts(BT::PortsList addition)
+  {
+    BT::PortsList basic = {
+      BT::InputPort<std::string>("server_name", "Action server name"),
+      BT::InputPort<std::chrono::milliseconds>("server_timeout")
+    };
+    basic.insert(addition.begin(), addition.end());
+
+    return basic;
+  }
+
+  /**
+   * @brief Creates list of BT ports
+   * @return BT::PortsList Containing basic ports along with node-specific ports
+   */
+  static BT::PortsList providedPorts()
+  {
+    return providedBasicPorts({});
+  }
+
+  /**
+   * @brief Function to perform some user-defined operation on tick
+   * Could do dynamic checks, such as getting updates to values on the blackboard
+   */
+  virtual void on_tick()
+  {
+  }
+
+  /**
+   * @brief The other (optional) override required by a BT action.
+   */
+  void halt() override
```

Comment:
I don't believe you have to override this if you have not state you need to reset